test/benchmark/iam/LanguageFeatureTest.java

```diff
@@ -0,0 +1,179 @@
+        benchmarker.setUp();
+        benchmarker.loadSchema("schema_types.tql");
+        benchmarker.loadSchema("schema_rules_optimised.tql");
+        benchmarker.loadSchema("schema_rules_test_specific.tql");
+        benchmarker.loadData("data.typedb");
+        benchmarker.warmUp();
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        benchmarker.tearDown();
+    }
+
+    @After
+    public void reset() {
+        benchmarker.reset();
+    }
+
+    @Test
+    public void testOwnership() {
+        String query = "match\n" +
+                "$x has parent-company-name $n;\n";
+        Benchmark benchmark = new Benchmark("ownership", query, 0); // TODO: Data only exists in the full data-set
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(100);
+        benchmark.assertCounters(50, 0, 2, 2);
+    }
+
+    @Test
+    public void testAttributeWithoutOwnership() {
+        String query = "match\n" +
+                "$n isa parent-company-name;\n";
+        Benchmark benchmark = new Benchmark("attribute-without-ownership", query, 0); // TODO: Data only exists in the full data-set
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(100);
+        benchmark.assertCounters(50, 0, 1, 1);
+    }
+
+    @Test
+    public void testRelationWithRolePlayers() {
+        String query = "match\n" +
+                "$ac (object: $obj, action: $action) isa access; \n" +
+                "$obj has id \"root/engineering/typedb/src\";" +
+                "$action has name \"delete file\";\n" +
+                "$p (subject: $s, access: $ac) isa permission;\n";
+        Benchmark benchmark = new Benchmark("relation-with-role-players", query, 10);
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(1000);
+        benchmark.assertCounters(500, 22, 30, 72);
+    }
+
+    @Test
+    public void testRelationWithoutRolePlayer() {
+        String query = "match\n" +
+                "$p isa permission;\n";
+        Benchmark benchmark = new Benchmark("relation-without-role-players", query, 2783);
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(5000);
+        benchmark.assertCounters(100, 2894, 800, 2500);
+    }
+
+    @Test
+    public void testInferredRelationAndOwnership() {
+        String query = "match\n" +
+                "$p (subject: $s, access: $ac) isa permission, has validity $v;\n";
+        Benchmark benchmark = new Benchmark("inferred-relation-and-ownership", query, 2783);
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(10_000);
+        benchmark.assertCounters(500, 5677, 10000, 15000);
```

Comment:
As another point, we get to get some high-level summaries of how much state we are creating and can assess that against if we think it's 'reasonable'. 15k streams for 2700 answers... That seems like a lot! Something we might look into in the future to see if we can reduce state somehow.

Reply:
There's one stream per branch in the conjunction. 
I already have a branch where we can re-use streams, but I don't know how widely applicable it is - we just need this benchmark done so I can justify the trade-off.

(Along with local-doneness, Trying to replace these streams with recompilation is a major goal for the reasoner rewrite)

Reply:
Materialisations + conjunction processors + compound streams is meant to reflect the memory usage. We're critically missing the intermediate answers created/buffered in the compound streams, which don't show up anywhere else.
I'll add that in somewhere.